activerecord/test/cases/associations/has_many_through_associations_test.rb

```diff
@@ -1258,32 +1258,6 @@ def test_has_many_through_obeys_order_on_through_association
     assert_equal ["parrot", "bulbul"], owner.toys.map { |r| r.pet.name }
   end
 
-  def test_has_many_through_associations_sum_on_columns
```

Comment:
Removed this test because it was really just testing `.distinct.sum` and has nothing to do with the `has_many` association.

Reply:
is this removal related to the Calculation#average?
If not related this should be address in a different PR, so the history get cleaner.
As usual, is maintainers call.

Reply:
I think it should be a separate PR as well. Looking at #566f1fd068711dfe557bef63406f8dd6d41d473d this test should ensure sum honors distinct on `has_many :through`. (It could use a better name though.. `test_has_many_through_associations_sum_distinct_on_columns`?)

Reply:
The commit message in 566f1fd068711dfe557bef63406f8dd6d41d473d  is actually misleading. What the changes introduced is actually to ensure that `distinct` works with `sum`. The `has_many :through` association has nothing to do with the changes made. I've split up the change in test in a separate commit.